src/legacy/core_plugins/vis_type_metric/public/components/metric_vis_component.tsx

```diff
@@ -70,79 +80,85 @@ export class MetricVisComponent extends Component {
     return bucket;
   }
 
-  _getColor(val, labels, colors) {
-    const bucket = this._getBucket(val);
+  private getColor(val: number, labels: string[], colors: any) {
+    const bucket = this.getBucket(val);
     const label = labels[bucket];
     return colors[label];
   }
 
-  _needsLightText(bgColor) {
-    const color = /rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)/.exec(bgColor);
-    if (!color) {
+  private needsLightText(bgColor: string) {
+    const colors = /rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)/.exec(bgColor);
+    if (!colors) {
       return false;
     }
-    return isColorDark(parseInt(color[1]), parseInt(color[2]), parseInt(color[3]));
+
+    const [red, green, blue] = colors.slice(1).map(parseInt);
+    return isColorDark(red, green, blue);
   }
 
-  _getFormattedValue = (fieldFormatter, value, format = 'text') => {
+  private getFormattedValue = (
+    fieldFormatter: FieldFormat,
+    value: any,
+    format: ContentType = 'text'
+  ) => {
     if (isNaN(value)) return '-';
     return fieldFormatter.convert(value, format);
   };
 
-  _processTableGroups(table) {
+  private processTableGroups(table: KibanaDatatable) {
     const config = this.props.visParams.metric;
     const dimensions = this.props.visParams.dimensions;
     const isPercentageMode = config.percentageMode;
     const min = config.colorsRange[0].from;
-    const max = last(config.colorsRange).to;
-    const colors = this._getColors();
-    const labels = this._getLabels();
-    const metrics = [];
+    const max = last<{ to: number }>(config.colorsRange).to;
+    const colors = this.getColors();
+    const labels = this.getLabels();
+    const metrics: any[] = []; // not yet typed
```

Comment:
It's fine to postpone these types, but could you create a "dummy type" for it? `export type MetricVisMetric = any` and then use `MetricVisMetric` everywhere. By doing so it will get much easier to propagate the type through your code once it's added.

Reply:
Great idea!